cluster/get-kube-binaries.sh

```diff
@@ -0,0 +1,188 @@
+#
+# We automatically choose the correct client binaries to download.
+#
+# Options:
+#  Set KUBERNETES_SERVER_ARCH to choose the server (Kubernetes cluster)
+#  architecture to download:
+#    * amd64 [default]
+#    * arm
+#    * arm64
+#    * ppc64le
+#
+#  Set KUBERNETES_SKIP_CONFIRM to skip the installation confirmation prompt.
+#  Set KUBERNETES_RELEASE_URL to choose where to download binaries from.
+#    (Default https://storage.googleapis.com/kubernetes-release/release).
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(cd $(dirname "${BASH_SOURCE}")/.. && pwd)
+
+KUBERNETES_RELEASE_URL="${KUBERNETES_RELEASE_URL:-https://storage.googleapis.com/kubernetes-release/release}"
+
+function detect_kube_release() {
+  if [[ ! -e "${KUBE_ROOT}/version" ]]; then
+    echo "Can't determine Kubernetes release." >&2
+    echo "This script should only be run from a prebuilt Kubernetes release." >&2
+    echo "Did you mean to use get-kube.sh instead?" >&2
+    exit 1
+  fi
+
+  KUBERNETES_RELEASE=$(cat "${KUBE_ROOT}/version")
+  DOWNLOAD_URL_PREFIX="${KUBERNETES_RELEASE_URL}/${KUBERNETES_RELEASE}"
+}
+
+function detect_client_info() {
+  local uname=$(uname)
+  if [[ "${uname}" == "Darwin" ]]; then
+    CLIENT_PLATFORM="darwin"
+  elif [[ "${uname}" == "Linux" ]]; then
+    CLIENT_PLATFORM="linux"
+  else
+    echo "Unknown, unsupported platform: (${uname})." >&2
+    echo "Supported platforms: Linux, Darwin." >&2
+    echo "Bailing out." >&2
+    exit 2
+  fi
+
+  local machine=$(uname -m)
+  if [[ "${machine}" == "x86_64" ]]; then
+    CLIENT_ARCH="amd64"
+  elif [[ "${machine}" == "i686" ]]; then
+    CLIENT_ARCH="386"
+  elif [[ "${machine}" == "arm" ]]; then
+    CLIENT_ARCH="arm"
+  elif [[ "${machine}" == "arm64" ]]; then
+    CLIENT_ARCH="arm64"
+  else
+    echo "Unknown, unsupported architecture (${machine})." >&2
+    echo "Supported architectures x86_64, i686, arm, arm64." >&2
+    echo "Bailing out." >&2
+    exit 3
+  fi
+}
+
+function md5sum_file() {
+  if which md5 >/dev/null 2>&1; then
+    md5 -q "$1"
+  else
+    md5sum "$1" | awk '{ print $1 }'
+  fi
+}
+
+function sha1sum_file() {
+  if which shasum >/dev/null 2>&1; then
+    shasum -a1 "$1" | awk '{ print $1 }'
+  else
+    sha1sum "$1" | awk '{ print $1 }'
+  fi
+}
+
+function download_tarball() {
+  local -r download_path="$1"
+  local -r file="$2"
+  url="${DOWNLOAD_URL_PREFIX}/${file}"
+  mkdir -p "${download_path}"
+  if [[ $(which curl) ]]; then
+    curl -L "${url}" -o "${download_path}/${file}"
+  elif [[ $(which wget) ]]; then
+    wget "${url}" -O "${download_path}/${file}"
+  else
+    echo "Couldn't find curl or wget.  Bailing out." >&2
+    exit 4
+  fi
+  echo
+  local md5sum=$(md5sum_file "${download_path}/${file}")
+  echo "md5sum(${file})=${md5sum}"
+  local sha1sum=$(sha1sum_file "${download_path}/${file}")
+  echo "sha1sum(${file})=${sha1sum}"
+  echo
+  # TODO: add actual verification
```

Comment:
The kubernetes.tar.gz doesn't have any embedded hashes right now. I plan to add them, but I'm trying to avoid any changes to build/ until #30787 is done.